tests/suites/os/tests/connectivity/index.js

```diff
@@ -0,0 +1,184 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+"use strict";
+
+const URL_TEST = "www.google.com";
+
+module.exports = {
+  title: "Connectivity tests",
+  tests: [
+    {
+      title: "Interface tests",
+      tests: ["wired", "wireless"].map((adaptor) => {
+        return {
+          title: `${adaptor.charAt(0).toUpperCase()}${adaptor.slice(1)} test`,
+          os: {
+            type: "object",
+            required: ["network"],
+            properties: {
+              network: {
+                type: "object",
+                required: [adaptor],
+                properties: {
+                  [adaptor]: {
+                    type: "boolean",
+                    const: true,
+                  },
+                },
+              },
+            },
+          },
+          run: async function (test) {
+            let testIface = "";
+            if (adaptor === "wireless") {
+              testIface = "wifi";
+            } else {
+              testIface = "ethernet";
+            }
+            const iface = await this.context
+              .get()
+              .worker.executeCommandInHostOS(
+                `nmcli d  | grep ' ${testIface} ' | grep connected | awk '{print $1}'`,
+                this.context.get().link
+              );
+
+            if (iface === "") {
+              throw new Error(`No ${testIface} interface found.`);
+            }
+
+            await this.context
+              .get()
+              .worker.executeCommandInHostOS(
+                `ping -c 10 -I ${iface} ${URL_TEST}`,
+                this.context.get().link
+              );
+
+            test.ok(true, `${URL_TEST} responded over ${testIface}`);
```

Comment:
`test.ok`? 
What value are we comparing `true` to or if `test.ok` checks for boolean logic, what function call is it testing?

Reply:
@vipulgupta2048 
As far as I can tell, `test.ok()` just checks that the arguement you give evaluates to true. In this case, if the test ever reaches this point, it will pass. If the URL fails to respond for whatever reason, then an error gests thrown in the previous piece of code, and the test fails - I agree this doesn't seem great however